jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ConnectionSettings.kt

```diff
@@ -0,0 +1,9 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import software.aws.toolkits.core.credentials.ToolkitCredentialsProvider
+import software.aws.toolkits.core.region.AwsRegion
+
+data class ConnectionSettings(val credentials: ToolkitCredentialsProvider?, val region: AwsRegion?)
```

Comment:
Hrmm - was hoping that we could have the properties be 'non-null' and that the connectionsettings itself would be nullable in places where that was required. The whole benefit of having a single object is you no-longer need to do the `if (region == null || credentials == null)` thing anymore.

Reply:
Okay, but we still need to expose the region/creds for things like switching. 

Got API recommendations?

Reply:
but do they need to be nullable?

Reply:
I think this is a step in the right direction - shouldn't block the PR on it - but I'd like to get to a point where we can do a single null-check for the 'connection' being there.

Reply:
If the user has no creds on their machine, we have no choice but make it nullable.

We could have getConnectionState()? and region?, credentials? methods.

Reply:
Pushed PR update to address your concerns